<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>mehenDi: Tree-based differrential testing for RNA-Seq — mehenDi • mehenDi</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="mehenDi: Tree-based differrential testing for RNA-Seq — mehenDi"><meta name="description" content="This package provides a tree based differential testing method for RNA-Seq
data, where the tree is obtained from TreeTerminus. The
candidate nodes can consist of both leaves and inner nodes, with
inner nodes comprising a subset of transcripts that exhibit high uncertainty."><meta property="og:description" content="This package provides a tree based differential testing method for RNA-Seq
data, where the tree is obtained from TreeTerminus. The
candidate nodes can consist of both leaves and inner nodes, with
inner nodes comprising a subset of transcripts that exhibit high uncertainty."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">mehenDi</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/mehenDi.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/NPSDC/mehenDi/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>mehenDi: Tree-based differrential testing for RNA-Seq</h1>
      <small class="dont-index">Source: <a href="https://github.com/NPSDC/mehenDi/blob/HEAD/R/mehenDi-package.R" class="external-link"><code>R/mehenDi-package.R</code></a>, <a href="https://github.com/NPSDC/mehenDi/blob/HEAD/R/mehenDi.R" class="external-link"><code>R/mehenDi.R</code></a></small>
      <div class="d-none name"><code>mehenDi.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This package provides a tree based differential testing method for RNA-Seq
data, where the tree is obtained from <strong>TreeTerminus</strong>. The
candidate nodes can consist of both leaves and inner nodes, with
inner nodes comprising a subset of transcripts that exhibit high uncertainty.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">mehenDi</span><span class="op">(</span></span>
<span>  <span class="va">tse</span>,</span>
<span>  <span class="va">x</span>,</span>
<span>  <span class="va">pvalues</span>,</span>
<span>  minP <span class="op">=</span> <span class="fl">0.7</span>,</span>
<span>  mIRVThresh <span class="op">=</span> <span class="fl">0.4</span>,</span>
<span>  alpha <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>  pThresh <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  cores <span class="op">=</span> <span class="fl">1</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-tse">tse<a class="anchor" aria-label="anchor" href="#arg-tse"></a></dt>
<dd><p>TreeSummarizedExperiment which contains the scaled inferential replicates</p></dd>


<dt id="arg-x">x<a class="anchor" aria-label="anchor" href="#arg-x"></a></dt>
<dd><p>character, the name of the condition variable. A factor for two group analysis</p></dd>


<dt id="arg-pvalues">pvalues<a class="anchor" aria-label="anchor" href="#arg-pvalues"></a></dt>
<dd><p>numeric, pvalues for all the nodes in the tree</p></dd>


<dt id="arg-minp">minP<a class="anchor" aria-label="anchor" href="#arg-minp"></a></dt>
<dd><p>numeric, value betweem 0 and 1, the proportion of the total inferential replicates
which should have the same sign change between conditions (default 0.7)</p></dd>


<dt id="arg-mirvthresh">mIRVThresh<a class="anchor" aria-label="anchor" href="#arg-mirvthresh"></a></dt>
<dd><p>numeric, minimum meanInfRV that a node should have for it to be considered for aggregation (default 0.4)</p></dd>


<dt id="arg-alpha">alpha<a class="anchor" aria-label="anchor" href="#arg-alpha"></a></dt>
<dd><p>numeric, the rate for the BH correction on the leaves which will be the pvalue threshold for deeming a node significant, wont be used if pThresh is set to some value</p></dd>


<dt id="arg-pthresh">pThresh<a class="anchor" aria-label="anchor" href="#arg-pthresh"></a></dt>
<dd><p>numeric, pvalue threshold for climbing, default is NULL since pThresh is computed automatically based on alpha, if not set to NULL then this pvalue threshold will used for deeming a node significant</p></dd>


<dt id="arg-cores">cores<a class="anchor" aria-label="anchor" href="#arg-cores"></a></dt>
<dd><p>numeric, the number of cores that will be used during parallelization</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>list that contains the selected nodes and the pvalue threshold used for deeeming
a node significant</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>The main function is:</p><ul><li><p><code>mehenDi</code> - performs tree-based differential testing</p></li>
</ul></div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package<span class="op">=</span><span class="st">"brainSimSmall"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">clustFile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">dir</span>, <span class="st">"cluster_nwk.txt"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">quantDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">dir</span>, <span class="st">"salmon_quants"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.vector.html" class="external-link">as.vector</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/outer.html" class="external-link">outer</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span>,<span class="va">y</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">x</span>,<span class="va">y</span>,sep<span class="op">=</span><span class="st">'_'</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">quantFiles</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">quantDir</span>, <span class="va">samples</span>, <span class="st">'quant.sf'</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">coldata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>files<span class="op">=</span><span class="va">quantFiles</span>, names<span class="op">=</span><span class="va">samples</span>, condition<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, each<span class="op">=</span><span class="fl">6</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu">beaveR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/beaveR/man/buildTSE.html" class="external-link">buildTSE</a></span><span class="op">(</span>treeTermFile <span class="op">=</span> <span class="va">clustFile</span>, coldata <span class="op">=</span> <span class="va">coldata</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> reading tree</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> importing quantifications</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> reading in files with read_tsv</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 3 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 4 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 5 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 6 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 7 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 8 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 9 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 10 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 11 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 12 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> found matching transcriptome:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> [ GENCODE - Homo sapiens - release 26 ]</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading existing TxDb created: 2023-03-02 17:08:07</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading existing transcript ranges created: 2023-03-02 21:45:10</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Missing txps 12</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Aggregation Started</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Aggregation Ended</span>
<span class="r-in"><span><span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu">TreeSummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/TreeSummarizedExperiment/man/TreeSummarizedExperiment-accessor.html" class="external-link">rowTree</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">l</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">tree</span><span class="op">$</span><span class="va">tip</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">yAll</span> <span class="op">&lt;-</span> <span class="fu">beaveR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/beaveR/man/computeSizeFactors.html" class="external-link">computeSizeFactors</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Progress:   1 on 100  Progress:   2 on 100  Progress:   3 on 100  Progress:   4 on 100  Progress:   5 on 100  Progress:   6 on 100  Progress:   7 on 100  Progress:   8 on 100  Progress:   9 on 100  Progress:  10 on 100  Progress:  11 on 100  Progress:  12 on 100  Progress:  13 on 100  Progress:  14 on 100  Progress:  15 on 100  Progress:  16 on 100  Progress:  17 on 100  Progress:  18 on 100  Progress:  19 on 100  Progress:  20 on 100  Progress:  21 on 100  Progress:  22 on 100  Progress:  23 on 100  Progress:  24 on 100  Progress:  25 on 100  Progress:  26 on 100  Progress:  27 on 100  Progress:  28 on 100  Progress:  29 on 100  Progress:  30 on 100  Progress:  31 on 100  Progress:  32 on 100  Progress:  33 on 100  Progress:  34 on 100  Progress:  35 on 100  Progress:  36 on 100  Progress:  37 on 100  Progress:  38 on 100  Progress:  39 on 100  Progress:  40 on 100  Progress:  41 on 100  Progress:  42 on 100  Progress:  43 on 100  Progress:  44 on 100  Progress:  45 on 100  Progress:  46 on 100  Progress:  47 on 100  Progress:  48 on 100  Progress:  49 on 100  Progress:  50 on 100  Progress:  51 on 100  Progress:  52 on 100  Progress:  53 on 100  Progress:  54 on 100  Progress:  55 on 100  Progress:  56 on 100  Progress:  57 on 100  Progress:  58 on 100  Progress:  59 on 100  Progress:  60 on 100  Progress:  61 on 100  Progress:  62 on 100  Progress:  63 on 100  Progress:  64 on 100  Progress:  65 on 100  Progress:  66 on 100  Progress:  67 on 100  Progress:  68 on 100  Progress:  69 on 100  Progress:  70 on 100  Progress:  71 on 100  Progress:  72 on 100  Progress:  73 on 100  Progress:  74 on 100  Progress:  75 on 100  Progress:  76 on 100  Progress:  77 on 100  Progress:  78 on 100  Progress:  79 on 100  Progress:  80 on 100  Progress:  81 on 100  Progress:  82 on 100  Progress:  83 on 100  Progress:  84 on 100  Progress:  85 on 100  Progress:  86 on 100  Progress:  87 on 100  Progress:  88 on 100  Progress:  89 on 100  Progress:  90 on 100  Progress:  91 on 100  Progress:  92 on 100  Progress:  93 on 100  Progress:  94 on 100  Progress:  95 on 100  Progress:  96 on 100  Progress:  97 on 100  Progress:  98 on 100  Progress:  99 on 100  Progress: 100 on 100  </span>
<span class="r-in"><span><span class="va">yAll</span> <span class="op">&lt;-</span> <span class="fu">beaveR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/beaveR/man/scInfReps.html" class="external-link">scInfReps</a></span><span class="op">(</span><span class="va">yAll</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Setting inf rep depth to mean depth</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Progress:   1 on 100  Progress:   2 on 100  Progress:   3 on 100  Progress:   4 on 100  Progress:   5 on 100  Progress:   6 on 100  Progress:   7 on 100  Progress:   8 on 100  Progress:   9 on 100  Progress:  10 on 100  Progress:  11 on 100  Progress:  12 on 100  Progress:  13 on 100  Progress:  14 on 100  Progress:  15 on 100  Progress:  16 on 100  Progress:  17 on 100  Progress:  18 on 100  Progress:  19 on 100  Progress:  20 on 100  Progress:  21 on 100  Progress:  22 on 100  Progress:  23 on 100  Progress:  24 on 100  Progress:  25 on 100  Progress:  26 on 100  Progress:  27 on 100  Progress:  28 on 100  Progress:  29 on 100  Progress:  30 on 100  Progress:  31 on 100  Progress:  32 on 100  Progress:  33 on 100  Progress:  34 on 100  Progress:  35 on 100  Progress:  36 on 100  Progress:  37 on 100  Progress:  38 on 100  Progress:  39 on 100  Progress:  40 on 100  Progress:  41 on 100  Progress:  42 on 100  Progress:  43 on 100  Progress:  44 on 100  Progress:  45 on 100  Progress:  46 on 100  Progress:  47 on 100  Progress:  48 on 100  Progress:  49 on 100  Progress:  50 on 100  Progress:  51 on 100  Progress:  52 on 100  Progress:  53 on 100  Progress:  54 on 100  Progress:  55 on 100  Progress:  56 on 100  Progress:  57 on 100  Progress:  58 on 100  Progress:  59 on 100  Progress:  60 on 100  Progress:  61 on 100  Progress:  62 on 100  Progress:  63 on 100  Progress:  64 on 100  Progress:  65 on 100  Progress:  66 on 100  Progress:  67 on 100  Progress:  68 on 100  Progress:  69 on 100  Progress:  70 on 100  Progress:  71 on 100  Progress:  72 on 100  Progress:  73 on 100  Progress:  74 on 100  Progress:  75 on 100  Progress:  76 on 100  Progress:  77 on 100  Progress:  78 on 100  Progress:  79 on 100  Progress:  80 on 100  Progress:  81 on 100  Progress:  82 on 100  Progress:  83 on 100  Progress:  84 on 100  Progress:  85 on 100  Progress:  86 on 100  Progress:  87 on 100  Progress:  88 on 100  Progress:  89 on 100  Progress:  90 on 100  Progress:  91 on 100  Progress:  92 on 100  Progress:  93 on 100  Progress:  94 on 100  Progress:  95 on 100  Progress:  96 on 100  Progress:  97 on 100  Progress:  98 on 100  Progress:  99 on 100  Progress: 100 on 100  </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span><span class="va">yAll</span> <span class="op">&lt;-</span> <span class="fu">fishpond</span><span class="fu">::</span><span class="fu"><a href="https://thelovelab.github.io/fishpond/reference/labelKeep.html" class="external-link">labelKeep</a></span><span class="op">(</span><span class="va">yAll</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">yTxps</span> <span class="op">&lt;-</span> <span class="fu">fishpond</span><span class="fu">::</span><span class="fu"><a href="https://thelovelab.github.io/fishpond/reference/swish.html" class="external-link">swish</a></span><span class="op">(</span><span class="va">yAll</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="va">l</span>,<span class="op">]</span>, x<span class="op">=</span><span class="st">"condition"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> generating test statistics over permutations</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Progress:   1 on 100  Progress:   2 on 100  Progress:   3 on 100  Progress:   4 on 100  Progress:   5 on 100  Progress:   6 on 100  Progress:   7 on 100  Progress:   8 on 100  Progress:   9 on 100  Progress:  10 on 100  Progress:  11 on 100  Progress:  12 on 100  Progress:  13 on 100  Progress:  14 on 100  Progress:  15 on 100  Progress:  16 on 100  Progress:  17 on 100  Progress:  18 on 100  Progress:  19 on 100  Progress:  20 on 100  Progress:  21 on 100  Progress:  22 on 100  Progress:  23 on 100  Progress:  24 on 100  Progress:  25 on 100  Progress:  26 on 100  Progress:  27 on 100  Progress:  28 on 100  Progress:  29 on 100  Progress:  30 on 100  Progress:  31 on 100  Progress:  32 on 100  Progress:  33 on 100  Progress:  34 on 100  Progress:  35 on 100  Progress:  36 on 100  Progress:  37 on 100  Progress:  38 on 100  Progress:  39 on 100  Progress:  40 on 100  Progress:  41 on 100  Progress:  42 on 100  Progress:  43 on 100  Progress:  44 on 100  Progress:  45 on 100  Progress:  46 on 100  Progress:  47 on 100  Progress:  48 on 100  Progress:  49 on 100  Progress:  50 on 100  Progress:  51 on 100  Progress:  52 on 100  Progress:  53 on 100  Progress:  54 on 100  Progress:  55 on 100  Progress:  56 on 100  Progress:  57 on 100  Progress:  58 on 100  Progress:  59 on 100  Progress:  60 on 100  Progress:  61 on 100  Progress:  62 on 100  Progress:  63 on 100  Progress:  64 on 100  Progress:  65 on 100  Progress:  66 on 100  Progress:  67 on 100  Progress:  68 on 100  Progress:  69 on 100  Progress:  70 on 100  Progress:  71 on 100  Progress:  72 on 100  Progress:  73 on 100  Progress:  74 on 100  Progress:  75 on 100  Progress:  76 on 100  Progress:  77 on 100  Progress:  78 on 100  Progress:  79 on 100  Progress:  80 on 100  Progress:  81 on 100  Progress:  82 on 100  Progress:  83 on 100  Progress:  84 on 100  Progress:  85 on 100  Progress:  86 on 100  Progress:  87 on 100  Progress:  88 on 100  Progress:  89 on 100  Progress:  90 on 100  Progress:  91 on 100  Progress:  92 on 100  Progress:  93 on 100  Progress:  94 on 100  Progress:  95 on 100  Progress:  96 on 100  Progress:  97 on 100  Progress:  98 on 100  Progress:  99 on 100  Progress: 100 on 100  </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span><span class="va">yInn</span> <span class="op">&lt;-</span> <span class="fu">fishpond</span><span class="fu">::</span><span class="fu"><a href="https://thelovelab.github.io/fishpond/reference/swish.html" class="external-link">swish</a></span><span class="op">(</span><span class="va">yAll</span><span class="op">[</span><span class="op">(</span><span class="va">l</span><span class="op">+</span><span class="fl">1</span><span class="op">)</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">yAll</span><span class="op">)</span>,<span class="op">]</span>, x<span class="op">=</span><span class="st">"condition"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> generating test statistics over permutations</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Progress:   1 on 100  Progress:   2 on 100  Progress:   3 on 100  Progress:   4 on 100  Progress:   5 on 100  Progress:   6 on 100  Progress:   7 on 100  Progress:   8 on 100  Progress:   9 on 100  Progress:  10 on 100  Progress:  11 on 100  Progress:  12 on 100  Progress:  13 on 100  Progress:  14 on 100  Progress:  15 on 100  Progress:  16 on 100  Progress:  17 on 100  Progress:  18 on 100  Progress:  19 on 100  Progress:  20 on 100  Progress:  21 on 100  Progress:  22 on 100  Progress:  23 on 100  Progress:  24 on 100  Progress:  25 on 100  Progress:  26 on 100  Progress:  27 on 100  Progress:  28 on 100  Progress:  29 on 100  Progress:  30 on 100  Progress:  31 on 100  Progress:  32 on 100  Progress:  33 on 100  Progress:  34 on 100  Progress:  35 on 100  Progress:  36 on 100  Progress:  37 on 100  Progress:  38 on 100  Progress:  39 on 100  Progress:  40 on 100  Progress:  41 on 100  Progress:  42 on 100  Progress:  43 on 100  Progress:  44 on 100  Progress:  45 on 100  Progress:  46 on 100  Progress:  47 on 100  Progress:  48 on 100  Progress:  49 on 100  Progress:  50 on 100  Progress:  51 on 100  Progress:  52 on 100  Progress:  53 on 100  Progress:  54 on 100  Progress:  55 on 100  Progress:  56 on 100  Progress:  57 on 100  Progress:  58 on 100  Progress:  59 on 100  Progress:  60 on 100  Progress:  61 on 100  Progress:  62 on 100  Progress:  63 on 100  Progress:  64 on 100  Progress:  65 on 100  Progress:  66 on 100  Progress:  67 on 100  Progress:  68 on 100  Progress:  69 on 100  Progress:  70 on 100  Progress:  71 on 100  Progress:  72 on 100  Progress:  73 on 100  Progress:  74 on 100  Progress:  75 on 100  Progress:  76 on 100  Progress:  77 on 100  Progress:  78 on 100  Progress:  79 on 100  Progress:  80 on 100  Progress:  81 on 100  Progress:  82 on 100  Progress:  83 on 100  Progress:  84 on 100  Progress:  85 on 100  Progress:  86 on 100  Progress:  87 on 100  Progress:  88 on 100  Progress:  89 on 100  Progress:  90 on 100  Progress:  91 on 100  Progress:  92 on 100  Progress:  93 on 100  Progress:  94 on 100  Progress:  95 on 100  Progress:  96 on 100  Progress:  97 on 100  Progress:  98 on 100  Progress:  99 on 100  Progress: 100 on 100  </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span><span class="va">pvals</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">yTxps</span><span class="op">)</span><span class="op">[[</span><span class="st">"pvalue"</span><span class="op">]</span><span class="op">]</span>, <span class="fu">mcols</span><span class="op">(</span><span class="va">yInn</span><span class="op">)</span><span class="op">[[</span><span class="st">"pvalue"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">tD</span> <span class="op">&lt;-</span> <span class="fu">mehenDi</span><span class="op">(</span><span class="va">yAll</span>, x<span class="op">=</span><span class="st">"condition"</span>, pvalues <span class="op">=</span> <span class="va">pvals</span>,</span></span>
<span class="r-in"><span>                   minP<span class="op">=</span><span class="fl">0.7</span>, mIRVThresh<span class="op">=</span><span class="fl">0.4</span>, alpha<span class="op">=</span><span class="fl">0.01</span><span class="op">)</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Noor Pratap Singh, Michael Love, Rob Pato.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

