<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Tree-based differential testing using inferential uncertainty for RNA-Seq â€¢ mehenDi</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Tree-based differential testing using inferential uncertainty for RNA-Seq">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">mehenDi</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/mehenDi.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/NPSDC/mehenDi/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Tree-based differential testing using inferential uncertainty for RNA-Seq</h1>
                        <h4 data-toc-skip class="author">Noor Pratap
Singh</h4>
            
            <h4 data-toc-skip class="date">02/12/2025</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/NPSDC/mehenDi/blob/HEAD/vignettes/mehenDi.Rmd" class="external-link"><code>vignettes/mehenDi.Rmd</code></a></small>
      <div class="d-none name"><code>mehenDi.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"NPSDC/mehenDi"</span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"NPSDC/brainSimSmall"</span><span class="op">)</span> <span class="co">## Download the package with a small dataset</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">mehenDi</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>We also install the package <em>brainSimSmall</em> which is a small
RNA-Seq dataset, on which this example is run.</p>
</div>
<div class="section level2">
<h2 id="mehendi">mehenDi<a class="anchor" aria-label="anchor" href="#mehendi"></a>
</h2>
<p>The <em>mehenDi</em> method performs differential testing use the
transcripts-tree obtained from <em>TreeTerminus</em>. The transcripts
are arranged in a way such that the uncertainty decreases on ascending
the tree, with the inner nodes comprising transcripts groups.
<em>mehenDi</em> aims to leverage the tree structure to find
differential signal in the tree which might get lost due to uncertainty.
The selected nodes can consist of both transcripts and inner nodes, with
inner nodes representing a set of transcripts that exhibit high
uncertainty.</p>
</div>
<div class="section level2">
<h2 id="quick-start">Quick start<a class="anchor" aria-label="anchor" href="#quick-start"></a>
</h2>
<p>We demonstrate <em>mehenDi</em> on a small simulated RNA-Seq dataset
in the <a href="https://github.com/NPSDC/brainSimSmall" class="external-link">brainSimSmall</a> package.
We have <em>Salmon</em> quantified counts for the samples, with 100
inferential replicates generated using Gibbs Sampling. <em>Note</em>
Gibbs Sampling can be replaced by bootstrap sampling. The dataset
consists of 12 samples belonging to two-groups, with 6 samples in
each.</p>
<div class="section level3">
<h3 id="loading-data">Loading data<a class="anchor" aria-label="anchor" href="#loading-data"></a>
</h3>
<p>We first load the data using <a href="https://github.com/NPSDC/beaveR" class="external-link">beaveR</a> to create
<em>TreeSummarizedExperiment</em> object. The leaves consist of the
transcripts.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package<span class="op">=</span><span class="st">"brainSimSmall"</span><span class="op">)</span></span>
<span><span class="va">clustFile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">dir</span>, <span class="st">"cluster_nwk.txt"</span><span class="op">)</span></span>
<span><span class="va">quantDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">dir</span>, <span class="st">"salmon_quants"</span><span class="op">)</span></span>
<span><span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.vector.html" class="external-link">as.vector</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/outer.html" class="external-link">outer</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span>,<span class="va">y</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">x</span>,<span class="va">y</span>,sep<span class="op">=</span><span class="st">'_'</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">quantFiles</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">quantDir</span>, <span class="va">samples</span>, <span class="st">'quant.sf'</span><span class="op">)</span></span>
<span><span class="va">coldata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>files<span class="op">=</span><span class="va">quantFiles</span>, names<span class="op">=</span><span class="va">samples</span>, condition<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, each<span class="op">=</span><span class="fl">6</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu">beaveR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/beaveR/man/buildTSE.html" class="external-link">buildTSE</a></span><span class="op">(</span>treeTermFile <span class="op">=</span> <span class="va">clustFile</span>, coldata <span class="op">=</span> <span class="va">coldata</span><span class="op">)</span></span>
<span><span class="va">tse</span></span>
<span><span class="co">#&gt; class: TreeSummarizedExperiment </span></span>
<span><span class="co">#&gt; dim: 8942 12 </span></span>
<span><span class="co">#&gt; metadata(8): tximetaInfo quantInfo ... txpsAnn infRepsScaled</span></span>
<span><span class="co">#&gt; assays(105): counts abundance ... mean variance</span></span>
<span><span class="co">#&gt; rownames(8942): ENST00000526904.5 ENST00000485271.5 ... Node8941</span></span>
<span><span class="co">#&gt;   Node8942</span></span>
<span><span class="co">#&gt; rowData names(1): meanInfRV</span></span>
<span><span class="co">#&gt; colnames(12): 1_1 2_1 ... 5_2 6_2</span></span>
<span><span class="co">#&gt; colData names(2): names condition</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span>
<span><span class="co">#&gt; rowLinks: a LinkDataFrame (8942 rows)</span></span>
<span><span class="co">#&gt; rowTree: 1 phylo tree(s) (4969 leaves)</span></span>
<span><span class="co">#&gt; colLinks: NULL</span></span>
<span><span class="co">#&gt; colTree: NULL</span></span>
<span><span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu">TreeSummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/TreeSummarizedExperiment/man/TreeSummarizedExperiment-accessor.html" class="external-link">rowTree</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span></span>
<span><span class="va">tree</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Phylogenetic tree with 4969 tips and 3973 internal nodes.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Tip labels:</span></span>
<span><span class="co">#&gt;   ENST00000526904.5, ENST00000485271.5, ENST00000635731.1, ENST00000636169.1, ENST00000532543.1, ENST00000483653.1, ...</span></span>
<span><span class="co">#&gt; Node labels:</span></span>
<span><span class="co">#&gt;   Node4970, Node4971, Node4972, Node4973, Node4974, Node4975, ...</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Unrooted; includes branch lengths.</span></span>
<span><span class="va">l</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">tree</span><span class="op">$</span><span class="va">tip</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="differential-transcript-expression">Differential transcript expression<a class="anchor" aria-label="anchor" href="#differential-transcript-expression"></a>
</h3>
<p>We first perform differential testing on the transcripts using Swish.
We can see the transcripts that are called significant at the 0.01 FDR
threshold.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">yTxps</span> <span class="op">&lt;-</span> <span class="va">tse</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="va">l</span>,<span class="op">]</span></span>
<span><span class="va">yTxps</span> <span class="op">&lt;-</span> <span class="fu">fishpond</span><span class="fu">::</span><span class="fu"><a href="https://thelovelab.github.io/fishpond/reference/scaleInfReps.html" class="external-link">scaleInfReps</a></span><span class="op">(</span><span class="va">yTxps</span>, quiet<span class="op">=</span><span class="cn">T</span><span class="op">)</span></span>
<span><span class="va">yTxps</span> <span class="op">&lt;-</span> <span class="fu">fishpond</span><span class="fu">::</span><span class="fu"><a href="https://thelovelab.github.io/fishpond/reference/labelKeep.html" class="external-link">labelKeep</a></span><span class="op">(</span><span class="va">yTxps</span><span class="op">)</span></span>
<span><span class="va">yTxps</span> <span class="op">&lt;-</span> <span class="fu">fishpond</span><span class="fu">::</span><span class="fu"><a href="https://thelovelab.github.io/fishpond/reference/swish.html" class="external-link">swish</a></span><span class="op">(</span><span class="va">yTxps</span>, x<span class="op">=</span><span class="st">"condition"</span>, quiet<span class="op">=</span><span class="cn">T</span><span class="op">)</span></span>
<span><span class="va">yTxps</span> <span class="op">&lt;-</span> <span class="va">yTxps</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">yTxps</span><span class="op">)</span>, <span class="va">tree</span><span class="op">$</span><span class="va">tip.label</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">yTxps</span><span class="op">)</span><span class="op">[[</span><span class="st">"qvalue"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] NA</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="running-mehendi">Running mehenDi<a class="anchor" aria-label="anchor" href="#running-mehendi"></a>
</h3>
<p>We next run <em>mehenDi</em> on the object <em>tse</em> created
above. The pipeline is similar to Swish, where first the data is
normalized (we used median-ratio scaling) in this example. It has been
broken down into two steps, in the first we compute the size factors and
then use the computed size factors for scaling the inferential
replicates. We have used <em>Swish</em> for computing the pvalues,
however we compute them separately for the leaves and the inner nodes.
This is because due to lower uncertainty, the width of the distribution
of the test-statistic of inner nodes is wider and thus have smaller
p-values compared to the inner nodes.</p>
<p><em>mehenDi</em> outputs a set of selected nodes based on a filtering
criteria. Once a node that has a p-value less than a certain threshold
(governed by (alpha)) is found, we check for different criteria. First,
nodes should consist of atleast one child node that has a p-value larger
than a threshold, otherwise we are overaggregating. We then check if all
the underlying descendant nodes that have a confident direction of
sign-change have the same direction change between conditions. The
confidence is determined by the (minP) parameters and looks at the
inferential replicates. Finally, we also want that the children nodes of
a selected node, to have uncertain abundance estimates (governed by
(meanInfRV)).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">mehenDi</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">yAll</span> <span class="op">&lt;-</span> <span class="fu">beaveR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/beaveR/man/computeSizeFactors.html" class="external-link">computeSizeFactors</a></span><span class="op">(</span><span class="va">tse</span>, quiet<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">yAll</span> <span class="op">&lt;-</span> <span class="fu">beaveR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/beaveR/man/scInfReps.html" class="external-link">scInfReps</a></span><span class="op">(</span><span class="va">yAll</span>, quiet<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">yAll</span> <span class="op">&lt;-</span> <span class="fu">fishpond</span><span class="fu">::</span><span class="fu"><a href="https://thelovelab.github.io/fishpond/reference/labelKeep.html" class="external-link">labelKeep</a></span><span class="op">(</span><span class="va">yAll</span><span class="op">)</span></span>
<span><span class="va">yInn</span> <span class="op">&lt;-</span> <span class="va">yAll</span><span class="op">[</span><span class="op">(</span><span class="va">l</span><span class="op">+</span><span class="fl">1</span><span class="op">)</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">yAll</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="va">yInn</span> <span class="op">&lt;-</span> <span class="fu">fishpond</span><span class="fu">::</span><span class="fu"><a href="https://thelovelab.github.io/fishpond/reference/swish.html" class="external-link">swish</a></span><span class="op">(</span><span class="va">yInn</span>, x<span class="op">=</span><span class="st">"condition"</span>, quiet<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pvals</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">yTxps</span><span class="op">)</span><span class="op">[[</span><span class="st">"pvalue"</span><span class="op">]</span><span class="op">]</span>, <span class="fu">mcols</span><span class="op">(</span><span class="va">yInn</span><span class="op">)</span><span class="op">[[</span><span class="st">"pvalue"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">tD</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mehenDi.html">mehenDi</a></span><span class="op">(</span><span class="va">yAll</span>, x<span class="op">=</span><span class="st">"condition"</span>, pvalues <span class="op">=</span> <span class="va">pvals</span>,</span>
<span>                    minP<span class="op">=</span><span class="fl">0.7</span>, mIRVThresh<span class="op">=</span><span class="fl">0.4</span>, alpha<span class="op">=</span><span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">tD</span><span class="op">$</span><span class="va">selNodes</span><span class="op">)</span> <span class="co">## Total number of selected nodes</span></span>
<span><span class="co">#&gt; [1] 80</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">tD</span><span class="op">$</span><span class="va">selNodes</span> <span class="op">&gt;</span> <span class="va">l</span><span class="op">)</span> <span class="co">## Number of inner nodes that are selected</span></span>
<span><span class="co">#&gt; [1] 16</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="differential-gene-expression">Differential gene expression<a class="anchor" aria-label="anchor" href="#differential-gene-expression"></a>
</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">yAllTxps</span> <span class="op">&lt;-</span> <span class="fu">tximeta</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/tximeta/man/tximeta.html" class="external-link">tximeta</a></span><span class="op">(</span><span class="va">coldata</span><span class="op">)</span></span>
<span><span class="va">gy</span> <span class="op">&lt;-</span> <span class="fu">tximeta</span><span class="fu">::</span><span class="fu">summarizeToGene</span><span class="op">(</span><span class="va">yAllTxps</span><span class="op">)</span></span>
<span><span class="va">gy</span> <span class="op">&lt;-</span> <span class="fu">fishpond</span><span class="fu">::</span><span class="fu"><a href="https://thelovelab.github.io/fishpond/reference/scaleInfReps.html" class="external-link">scaleInfReps</a></span><span class="op">(</span><span class="va">gy</span>, quiet<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">gy</span> <span class="op">&lt;-</span> <span class="fu">fishpond</span><span class="fu">::</span><span class="fu"><a href="https://thelovelab.github.io/fishpond/reference/labelKeep.html" class="external-link">labelKeep</a></span><span class="op">(</span><span class="va">gy</span><span class="op">)</span></span>
<span><span class="va">gy</span> <span class="op">&lt;-</span> <span class="fu">fishpond</span><span class="fu">::</span><span class="fu"><a href="https://thelovelab.github.io/fishpond/reference/swish.html" class="external-link">swish</a></span><span class="op">(</span><span class="va">gy</span>, x<span class="op">=</span><span class="st">"condition"</span>, quiet<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">gy</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 57808    12</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">gy</span><span class="op">)</span><span class="op">[[</span><span class="st">"qvalue"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; FALSE  TRUE </span></span>
<span><span class="co">#&gt;    88    13</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="exploration-of-a-node-output-by-mehendi">Exploration of a node output by mehenDi<a class="anchor" aria-label="anchor" href="#exploration-of-a-node-output-by-mehendi"></a>
</h3>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nodeInf</span> <span class="op">&lt;-</span> <span class="fu">beaveR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/beaveR/man/findNodeInformation.html" class="external-link">findNodeInformation</a></span><span class="op">(</span><span class="va">yAll</span>, node <span class="op">=</span> <span class="va">tD</span><span class="op">[[</span><span class="st">"selNodes"</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, type <span class="op">=</span> <span class="st">"tips"</span><span class="op">)</span></span>
<span><span class="va">nodeInf</span></span>
<span><span class="co">#&gt;                   nodeInd  meanInfRV        genes nodeType</span></span>
<span><span class="co">#&gt; ENST00000566784.5       9 0.05741234 ENSG0000....     Leaf</span></span>
<span></span>
<span><span class="co">### Distribution of the height of the nodes output by mehenDi</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/node.depth.html" class="external-link">node.depth</a></span><span class="op">(</span><span class="va">tree</span>, <span class="fl">2</span><span class="op">)</span><span class="op">[</span><span class="va">tD</span><span class="op">[[</span><span class="st">"selNodes"</span><span class="op">]</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  1  2  3  4  6  7  9 10 11 </span></span>
<span><span class="co">#&gt; 64  5  1  3  1  3  1  1  1</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="visualzing-the-transcript-trees-for-the-selected-nodes-output-by-mehendi">Visualzing the transcript trees for the selected nodes output by
mehenDi<a class="anchor" aria-label="anchor" href="#visualzing-the-transcript-trees-for-the-selected-nodes-output-by-mehendi"></a>
</h3>
<p>We use the library <code>plotGardner</code> for plotting the
transcripts of a gene and <code>ggtree</code> for the corresponding
trees</p>
<div class="section level4">
<h4 id="loading-packages-and-preparing-data-for-plotting">Loading packages and preparing data for plotting<a class="anchor" aria-label="anchor" href="#loading-packages-and-preparing-data-for-plotting"></a>
</h4>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">TxDb.Hsapiens.UCSC.hg38.knownGene</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">plyranges</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://www.amazon.com/Integration-Manipulation-Visualization-Phylogenetic-Computational-ebook/dp/B0B5NLZR1Z/" class="external-link">ggtree</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"grid"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/GuangchuangYu/ggplotify" class="external-link">"ggplotify"</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org" class="external-link">"ggplot2"</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://rpkgs.datanovia.com/ggpubr/" class="external-link">"ggpubr"</a></span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">txdb</span> <span class="op">&lt;-</span> <span class="va">TxDb.Hsapiens.UCSC.hg38.knownGene</span></span>
<span><span class="va">txpHuman</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">transcripts</a></span><span class="op">(</span><span class="va">txdb</span><span class="op">)</span></span>
<span></span>
<span><span class="va">txpHuman</span> <span class="op">&lt;-</span> <span class="va">txpHuman</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>tx_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">tx_id</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">txpHuman</span> <span class="op">&lt;-</span> <span class="va">txpHuman</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>gene_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">mapIds</a></span><span class="op">(</span></span>
<span>           <span class="va">txdb</span>, keys<span class="op">=</span><span class="va">tx_id</span>,</span>
<span>           column<span class="op">=</span><span class="st">"GENEID"</span>, keytype<span class="op">=</span><span class="st">"TXID"</span><span class="op">)</span></span>
<span>         <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">gene_id</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">txpHuman</span> <span class="op">&lt;-</span> <span class="va">txpHuman</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>ensID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">mapIds</a></span><span class="op">(</span></span>
<span>           <span class="va">org.Hs.eg.db</span>, keys<span class="op">=</span><span class="va">gene_id</span>,</span>
<span>           column<span class="op">=</span><span class="st">"ENSEMBL"</span>, keytype<span class="op">=</span><span class="st">"ENTREZID"</span><span class="op">)</span></span>
<span>         <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">ensID</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">txpHuman</span> <span class="op">&lt;-</span> <span class="va">txpHuman</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>                <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">tx_name</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">tree</span><span class="op">$</span><span class="va">tip</span>, <span class="fu">mcols</span><span class="op">(</span><span class="va">txpHuman</span><span class="op">)</span><span class="op">[[</span><span class="st">"tx_name"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">genesHuman</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">genes</a></span><span class="op">(</span><span class="va">txdb</span><span class="op">)</span></span>
<span><span class="va">genesHuman</span> <span class="op">&lt;-</span> <span class="va">genesHuman</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>ensID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">mapIds</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>,</span>
<span>                            <span class="va">gene_id</span>, <span class="st">"ENSEMBL"</span>, <span class="st">"ENTREZID"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">ensID</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="picking-a-node">Picking a node<a class="anchor" aria-label="anchor" href="#picking-a-node"></a>
</h4>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">node</span> <span class="op">&lt;-</span> <span class="va">tD</span><span class="op">$</span><span class="va">selNodes</span><span class="op">[</span><span class="va">tD</span><span class="op">$</span><span class="va">selNodes</span> <span class="op">&gt;</span> <span class="va">l</span><span class="op">]</span><span class="op">[</span><span class="fl">10</span><span class="op">]</span></span>
<span><span class="va">treeSub</span> <span class="op">&lt;-</span> <span class="fu">tidytree</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/tidytree/man/tree_subset.html" class="external-link">tree_subset</a></span><span class="op">(</span><span class="va">tree</span>, <span class="va">node</span>, levels_back <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span><span class="va">txShow</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">[</span><span class="fu">phangorn</span><span class="fu">::</span><span class="fu"><a href="https://klausvigo.github.io/phangorn/reference/Ancestors.html" class="external-link">Descendants</a></span><span class="op">(</span><span class="va">tree</span>,<span class="va">node</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="va">gs</span> <span class="op">&lt;-</span> <span class="va">txpHuman</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">tx_name</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">treeSub</span><span class="op">$</span><span class="va">tip</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>        <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>        <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">ensID</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>        <span class="va">unlist</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>        <span class="va">unique</span></span>
<span></span>
<span><span class="va">gTxps</span> <span class="op">&lt;-</span> <span class="va">txpHuman</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">ensID</span> <span class="op">==</span> <span class="va">gs</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>        <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>        <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">tx_name</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">tx_name</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">txShow</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>        <span class="va">unlist</span></span>
<span></span>
<span><span class="va">g</span> <span class="op">&lt;-</span> <span class="va">genesHuman</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">ensID</span> <span class="op">==</span> <span class="va">gs</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>        <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">txpMin</span> <span class="op">&lt;-</span> <span class="va">gTxps</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.min.html" class="external-link">which.min</a></span><span class="op">(</span><span class="fu">mcols</span><span class="op">(</span><span class="va">yTxps</span><span class="op">)</span><span class="op">[</span><span class="va">gTxps</span>, <span class="st">"pvalue"</span><span class="op">]</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">minTInd</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">txpMin</span>, <span class="va">treeSub</span><span class="op">$</span><span class="va">tip</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="extracting-plot-for-inferential-replicates">Extracting plot for inferential replicates<a class="anchor" aria-label="anchor" href="#extracting-plot-for-inferential-replicates"></a>
</h4>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cex</span> <span class="op">&lt;-</span> <span class="fl">1.6</span></span>
<span><span class="va">legPos</span> <span class="op">&lt;-</span> <span class="st">"topright"</span></span>
<span><span class="va">pTxp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggplotify/man/as-grob.html" class="external-link">as.grob</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="op">)</span> <span class="fu">fishpond</span><span class="fu">::</span><span class="fu"><a href="https://thelovelab.github.io/fishpond/reference/plotInfReps.html" class="external-link">plotInfReps</a></span><span class="op">(</span><span class="va">yAll</span>, <span class="va">txpMin</span>, x <span class="op">=</span> <span class="st">"condition"</span>, legend<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                              main<span class="op">=</span><span class="va">txpMin</span>, legendTitle<span class="op">=</span><span class="cn">TRUE</span>, legendCex<span class="op">=</span><span class="va">cex</span>, </span>
<span>                             legendPos <span class="op">=</span> <span class="va">legPos</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pInn</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggplotify/man/as-grob.html" class="external-link">as.grob</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="op">)</span> <span class="fu">fishpond</span><span class="fu">::</span><span class="fu"><a href="https://thelovelab.github.io/fishpond/reference/plotInfReps.html" class="external-link">plotInfReps</a></span><span class="op">(</span><span class="va">yAll</span>, <span class="va">node</span>, x <span class="op">=</span> <span class="st">"condition"</span>, legend<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                              main<span class="op">=</span><span class="va">txpMin</span>, legendTitle<span class="op">=</span><span class="cn">TRUE</span>, legendCex<span class="op">=</span><span class="va">cex</span>, </span>
<span>                             legendPos <span class="op">=</span> <span class="va">legPos</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="extracting-plot-for-tree">Extracting plot for tree<a class="anchor" aria-label="anchor" href="#extracting-plot-for-tree"></a>
</h4>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nI</span> <span class="op">&lt;-</span> <span class="fl">5</span></span>
<span><span class="va">size</span> <span class="op">&lt;-</span> <span class="fl">5</span></span>
<span><span class="va">fs</span> <span class="op">&lt;-</span> <span class="fl">20</span></span>
<span><span class="va">pTree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html" class="external-link">ggtree</a></span><span class="op">(</span><span class="va">treeSub</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html" class="external-link">xlim</a></span><span class="op">(</span><span class="cn">NA</span>, <span class="fl">80</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html" class="external-link">geom_tiplab</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">6</span>, hjust<span class="op">=</span><span class="op">-</span><span class="fl">0.1</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_point2.html" class="external-link">geom_point2</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>subset<span class="op">=</span><span class="op">(</span><span class="va">node</span><span class="op">==</span><span class="va">nI</span><span class="op">)</span>, color<span class="op">=</span><span class="st">"red"</span><span class="op">)</span>,</span>
<span>                size<span class="op">=</span><span class="fl">5</span>, fill<span class="op">=</span><span class="st">'red'</span>, show.legend<span class="op">=</span><span class="cn">T</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_point2.html" class="external-link">geom_point2</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>subset<span class="op">=</span><span class="op">(</span><span class="va">node</span><span class="op">==</span><span class="va">minTInd</span><span class="op">)</span>, color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,</span>
<span>                size<span class="op">=</span><span class="fl">5</span>, fill<span class="op">=</span><span class="st">'black'</span>, show.legend<span class="op">=</span><span class="cn">T</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_cladelab.html" class="external-link">geom_cladelab</a></span><span class="op">(</span>node <span class="op">=</span> <span class="va">nI</span>, label <span class="op">=</span> <span class="st">""</span>, textcolour<span class="op">=</span><span class="st">"red"</span>, barsize<span class="op">=</span><span class="fl">2</span>,</span>
<span>                      barcolour<span class="op">=</span><span class="st">"red"</span>,  fontsize<span class="op">=</span><span class="fl">5</span>, offset <span class="op">=</span> <span class="fl">36</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"bottom"</span>, legend.text<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size<span class="op">=</span><span class="va">fs</span><span class="op">)</span>,</span>
<span>             plot.title<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size<span class="op">=</span><span class="va">fs</span>, face<span class="op">=</span><span class="st">"bold"</span>, hjust<span class="op">=</span><span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">""</span>, labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Transcript with the lowest p-value"</span>, <span class="st">"Selected Node"</span><span class="op">)</span>,</span>
<span>               values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"black"</span>, <span class="st">"red"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="putting-it-all-together">Putting it all together<a class="anchor" aria-label="anchor" href="#putting-it-all-together"></a>
</h4>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.3.2 (2023-10-31)</span></span>
<span><span class="co">#&gt; Platform: x86_64-conda-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Red Hat Enterprise Linux 8.10 (Ootpa)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS/LAPACK: /fs/cbcb-lab/rob/students/noor/miniforge3/envs/R4.3/lib/libopenblasp-r0.3.25.so;  LAPACK version 3.11.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: America/New_York</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] grid      stats4    stats     graphics  grDevices utils     datasets </span></span>
<span><span class="co">#&gt; [8] methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] ggpubr_0.6.0                            </span></span>
<span><span class="co">#&gt;  [2] ggplot2_3.5.1                           </span></span>
<span><span class="co">#&gt;  [3] ggplotify_0.1.2                         </span></span>
<span><span class="co">#&gt;  [4] ggtree_3.10.1                           </span></span>
<span><span class="co">#&gt;  [5] plyranges_1.22.0                        </span></span>
<span><span class="co">#&gt;  [6] TxDb.Hsapiens.UCSC.hg38.knownGene_3.18.0</span></span>
<span><span class="co">#&gt;  [7] org.Hs.eg.db_3.18.0                     </span></span>
<span><span class="co">#&gt;  [8] mehenDi_0.99.0                          </span></span>
<span><span class="co">#&gt;  [9] GenomicFeatures_1.54.4                  </span></span>
<span><span class="co">#&gt; [10] AnnotationDbi_1.64.1                    </span></span>
<span><span class="co">#&gt; [11] Biobase_2.62.0                          </span></span>
<span><span class="co">#&gt; [12] GenomicRanges_1.54.1                    </span></span>
<span><span class="co">#&gt; [13] GenomeInfoDb_1.38.8                     </span></span>
<span><span class="co">#&gt; [14] IRanges_2.36.0                          </span></span>
<span><span class="co">#&gt; [15] S4Vectors_0.40.2                        </span></span>
<span><span class="co">#&gt; [16] BiocGenerics_0.48.1                     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] jsonlite_1.8.8                  tximport_1.30.0                </span></span>
<span><span class="co">#&gt;   [3] magrittr_2.0.3                  rmarkdown_2.27                 </span></span>
<span><span class="co">#&gt;   [5] fs_1.6.4                        BiocIO_1.12.0                  </span></span>
<span><span class="co">#&gt;   [7] zlibbioc_1.48.2                 ragg_1.3.2                     </span></span>
<span><span class="co">#&gt;   [9] vctrs_0.6.5                     memoise_2.0.1                  </span></span>
<span><span class="co">#&gt;  [11] Rsamtools_2.18.0                RCurl_1.98-1.16                </span></span>
<span><span class="co">#&gt;  [13] rstatix_0.7.2                   htmltools_0.5.8.1              </span></span>
<span><span class="co">#&gt;  [15] S4Arrays_1.2.1                  progress_1.2.3                 </span></span>
<span><span class="co">#&gt;  [17] AnnotationHub_3.10.1            curl_5.2.1                     </span></span>
<span><span class="co">#&gt;  [19] broom_1.0.6                     gridGraphics_0.5-1             </span></span>
<span><span class="co">#&gt;  [21] SparseArray_1.2.4               sass_0.4.9                     </span></span>
<span><span class="co">#&gt;  [23] bslib_0.7.0                     htmlwidgets_1.6.4              </span></span>
<span><span class="co">#&gt;  [25] desc_1.4.3                      plyr_1.8.9                     </span></span>
<span><span class="co">#&gt;  [27] cachem_1.1.0                    GenomicAlignments_1.38.2       </span></span>
<span><span class="co">#&gt;  [29] igraph_2.0.3                    mime_0.12                      </span></span>
<span><span class="co">#&gt;  [31] lifecycle_1.0.4                 pkgconfig_2.0.3                </span></span>
<span><span class="co">#&gt;  [33] Matrix_1.6-5                    R6_2.5.1                       </span></span>
<span><span class="co">#&gt;  [35] fastmap_1.2.0                   GenomeInfoDbData_1.2.11        </span></span>
<span><span class="co">#&gt;  [37] MatrixGenerics_1.14.0           shiny_1.8.1.1                  </span></span>
<span><span class="co">#&gt;  [39] aplot_0.2.3                     digest_0.6.36                  </span></span>
<span><span class="co">#&gt;  [41] colorspace_2.1-1                patchwork_1.2.0                </span></span>
<span><span class="co">#&gt;  [43] tximeta_1.20.3                  textshaping_0.4.0              </span></span>
<span><span class="co">#&gt;  [45] RSQLite_2.3.7                   filelock_1.0.3                 </span></span>
<span><span class="co">#&gt;  [47] fansi_1.0.6                     httr_1.4.7                     </span></span>
<span><span class="co">#&gt;  [49] TreeSummarizedExperiment_2.10.0 abind_1.4-5                    </span></span>
<span><span class="co">#&gt;  [51] compiler_4.3.2                  withr_3.0.0                    </span></span>
<span><span class="co">#&gt;  [53] bit64_4.0.5                     backports_1.5.0                </span></span>
<span><span class="co">#&gt;  [55] BiocParallel_1.36.0             carData_3.0-5                  </span></span>
<span><span class="co">#&gt;  [57] DBI_1.2.3                       ggsignif_0.6.4                 </span></span>
<span><span class="co">#&gt;  [59] biomaRt_2.58.2                  rappdirs_0.3.3                 </span></span>
<span><span class="co">#&gt;  [61] DelayedArray_0.28.0             rjson_0.2.23                   </span></span>
<span><span class="co">#&gt;  [63] gtools_3.9.5                    tools_4.3.2                    </span></span>
<span><span class="co">#&gt;  [65] ape_5.8                         interactiveDisplayBase_1.40.0  </span></span>
<span><span class="co">#&gt;  [67] httpuv_1.6.15                   glue_1.7.0                     </span></span>
<span><span class="co">#&gt;  [69] quadprog_1.5-8                  restfulr_0.0.15                </span></span>
<span><span class="co">#&gt;  [71] nlme_3.1-165                    promises_1.3.0                 </span></span>
<span><span class="co">#&gt;  [73] reshape2_1.4.4                  generics_0.1.3                 </span></span>
<span><span class="co">#&gt;  [75] gtable_0.3.5                    tzdb_0.4.0                     </span></span>
<span><span class="co">#&gt;  [77] tidyr_1.3.1                     ensembldb_2.26.0               </span></span>
<span><span class="co">#&gt;  [79] hms_1.1.3                       car_3.1-2                      </span></span>
<span><span class="co">#&gt;  [81] xml2_1.3.6                      utf8_1.2.4                     </span></span>
<span><span class="co">#&gt;  [83] XVector_0.42.0                  fishpond_2.8.0                 </span></span>
<span><span class="co">#&gt;  [85] BiocVersion_3.18.1              pillar_1.9.0                   </span></span>
<span><span class="co">#&gt;  [87] stringr_1.5.1                   vroom_1.6.5                    </span></span>
<span><span class="co">#&gt;  [89] yulab.utils_0.1.5               later_1.3.2                    </span></span>
<span><span class="co">#&gt;  [91] splines_4.3.2                   dplyr_1.1.4                    </span></span>
<span><span class="co">#&gt;  [93] BiocFileCache_2.10.2            treeio_1.26.0                  </span></span>
<span><span class="co">#&gt;  [95] lattice_0.22-6                  rtracklayer_1.62.0             </span></span>
<span><span class="co">#&gt;  [97] bit_4.0.5                       tidyselect_1.2.1               </span></span>
<span><span class="co">#&gt;  [99] SingleCellExperiment_1.24.0     Biostrings_2.70.3              </span></span>
<span><span class="co">#&gt; [101] beaveR_0.99.1                   knitr_1.48                     </span></span>
<span><span class="co">#&gt; [103] ProtGenerics_1.34.0             SummarizedExperiment_1.32.0    </span></span>
<span><span class="co">#&gt; [105] xfun_0.46                       matrixStats_1.3.0              </span></span>
<span><span class="co">#&gt; [107] stringi_1.8.4                   ggfun_0.1.5                    </span></span>
<span><span class="co">#&gt; [109] lazyeval_0.2.2                  yaml_2.3.10                    </span></span>
<span><span class="co">#&gt; [111] evaluate_0.24.0                 codetools_0.2-20               </span></span>
<span><span class="co">#&gt; [113] qvalue_2.34.0                   tibble_3.2.1                   </span></span>
<span><span class="co">#&gt; [115] BiocManager_1.30.23             cli_3.6.3                      </span></span>
<span><span class="co">#&gt; [117] xtable_1.8-4                    systemfonts_1.1.0              </span></span>
<span><span class="co">#&gt; [119] munsell_0.5.1                   jquerylib_0.1.4                </span></span>
<span><span class="co">#&gt; [121] Rcpp_1.0.13                     svMisc_1.2.3                   </span></span>
<span><span class="co">#&gt; [123] dbplyr_2.5.0                    png_0.1-8                      </span></span>
<span><span class="co">#&gt; [125] XML_3.99-0.17                   parallel_4.3.2                 </span></span>
<span><span class="co">#&gt; [127] pkgdown_2.1.1                   readr_2.1.5                    </span></span>
<span><span class="co">#&gt; [129] blob_1.2.4                      prettyunits_1.2.0              </span></span>
<span><span class="co">#&gt; [131] AnnotationFilter_1.26.0         bitops_1.0-7                   </span></span>
<span><span class="co">#&gt; [133] phangorn_2.11.1                 tidytree_0.4.6                 </span></span>
<span><span class="co">#&gt; [135] scales_1.3.0                    purrr_1.0.2                    </span></span>
<span><span class="co">#&gt; [137] crayon_1.5.3                    rlang_1.1.4                    </span></span>
<span><span class="co">#&gt; [139] fastmatch_1.1-4                 KEGGREST_1.42.0</span></span></code></pre></div>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Noor Pratap Singh, Michael Love, Rob Pato.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
